scipy/signal/_signaltools.py

```diff
@@ -3500,38 +3506,58 @@ def invresz(r, p, k, tol=1e-3, rtype='avg'):
-    """
-    Resample `x` to `num` samples using Fourier method along the given axis.
+    """Resample `x` to `num` samples using the Fourier method along the given `axis`.
 
-    The resampled signal starts at the same value as `x` but is sampled
-    with a spacing of ``len(x) / num * (spacing of x)``.  Because a
-    Fourier method is used, the signal is assumed to be periodic.
+    The resampling is performed by shortening or zero-padding the FFT of `x`. This has
+    the advantages of providing an ideal antialiasing filter and allowing arbitrary
+    up- or down-sampling ratios. The main drawback is the requirement of assuming `x`
+    to be a periodic signal.
 
     Parameters
     ----------
     x : array_like
-        The data to be resampled.
+        The input signal made up of equidistant samples. If `x` is a multidimensional
+        array, the parameter `axis` specifies the time/frequency axis. It is assumed
+        here that ``n_x = x.shape[axis]`` specifies the number of samples and ``T`` the
+        sampling interval.
     num : int
-        The number of samples in the resampled signal.
+        The number of samples of the resampled output signal. It may be larger or
+        smaller than ``n_x``.
     t : array_like, optional
-        If `t` is given, it is assumed to be the equally spaced sample
-        positions associated with the signal data in `x`.
+        If `t` is not ``None`` (default), then the timestamps of the resampled signal
+        are also returned. `t` must contain at least the first two timestamps of the
+        input signal `x` (all others are ignored). The timestamps of the output signal
+        are determined by ``t[0] + T * n_x / num * np.arange(num)`` with
+        ``T = t[1] - t[0]``.
     axis : int, optional
-        The axis of `x` that is resampled.  Default is 0.
+        The time/frequency axis of `x` along which the resampling take place.
+        The Default is 0.
     window : array_like, callable, string, float, or tuple, optional
-        Specifies the window applied to the signal in the Fourier
-        domain.  See below for details.
-    domain : string, optional
-        A string indicating the domain of the input `x`:
-        ``time`` Consider the input `x` as time-domain (Default),
-        ``freq`` Consider the input `x` as frequency-domain.
+        If not ``None`` (default), specifies a filter in the Fourier domain, which is
+        applied before resampling. I.e., the FFT ``X`` of `x` is calculated by
+        ``X = W * fft(x, axis=axis)``. ``W`` may be interpreted as a sepctral
+        windowing function ``W(f_X)`` which consumes the frequencies
+        ``f_X = fftfreq(n_x, T)``.
+
+        If `window` is a 1d array of length `n_x` then ``W=window``.
+        If `window` is a function  then ``W = window(f_X)``.
+        Otherwise, `window` is passed to `~scipy.signal.get_window`, i.e.,
+        ``W = fftshift(signal.get_window(window, n_x))``.
```

Comment:
Just to double-check: is `fftshift` not extraneous here?

Reply:
It is not, since it is used in the code as well. The reason is  that you typically want your spectral window to be centered over the zero frequency bin. For the standard `fft()`, the first entry corresponds to the zero frequency bin, hence `fftshift` is needed.